File: src/main/java/chloe/movietalk/domain/enums/Gender.java
package chloe.movietalk.domain.enums;

import chloe.movietalk.exception.global.InvalidGenderEnumValueException;

import java.util.Arrays;

public enum Gender {
    MALE,
    FEMALE,
    OTHER;

    public static Gender from(String genderStr) {
        if (genderStr == null)
            return null;
        return Arrays.stream(values())
                .filter(g -> g.name().equals(genderStr))
                .findFirst()
                .orElseThrow(() -> InvalidGenderEnumValueException.EXCEPTION);
    }
}

File: src/main/java/chloe/movietalk/domain/enums/Gender.kt
package chloe.movietalk.domain.enums

import chloe.movietalk.exception.global.InvalidGenderEnumValueException

enum class Gender {
    MALE,
    FEMALE,
    OTHER;

    companion object {
        fun from(genderStr: String?): Gender {
            return entries.find { it.name == genderStr }
                ?: throw InvalidGenderEnumValueException
        }
    }
}
